src/rest/AnalyticsEndpoint.ts

```diff
@@ -101,36 +103,46 @@ export class AnalyticsEndpoint {
         visitorId: Cookie.get('visitorId')
       }
     });
-
     // We use pendingRequest because we don't want to have 2 request to analytics at the same time.
     // Otherwise the cookie visitId won't be set correctly.
-    if (AnalyticsEndpoint.pendingRequest == null) {
-      AnalyticsEndpoint.pendingRequest = this.endpointCaller
-        .call<R>({
-          errorsAsSuccess: false,
-          method: 'POST',
-          queryString: urlNormalized.queryNormalized,
-          requestData: data,
-          url: urlNormalized.path,
-          responseType: 'text',
-          requestDataType: 'application/json'
-        })
-        .then((res: ISuccessResponse<R>) => {
-          return this.handleAnalyticsEventResponse(<any>res.data);
-        })
-        .finally(() => {
-          AnalyticsEndpoint.pendingRequest = null;
-        });
-      return AnalyticsEndpoint.pendingRequest;
-    } else {
-      return AnalyticsEndpoint.pendingRequest.finally(() => {
+    if (AnalyticsEndpoint.pendingRequest != null) {
+      try {
+        await AnalyticsEndpoint.pendingRequest;
+      } finally {
         return this.sendToService<D, R>(data, path, paramName);
-      });
+      }
+    }
+
+    const request: Promise<any> = (AnalyticsEndpoint.pendingRequest = this.endpointCaller.call<R>({
+      errorsAsSuccess: false,
+      method: 'POST',
+      queryString: urlNormalized.queryNormalized,
+      requestData: data,
+      url: urlNormalized.path,
+      responseType: 'text',
+      requestDataType: 'application/json'
+    }));
+
+    try {
+      const results = await request;
+      AnalyticsEndpoint.pendingRequest = null;
```

Comment:
You could add this line to a `finally` block of the `this.endpointCaller.call` promise to cut the repetition here and on line 133.

Reply:
Unfortunately no, since the finally blocks will get executed "sometimes later", as it it the behaviour of await/yield. I need the variable to be set to null as soon as it resolves so we can queue up another one if needed. Otherwise, other subsequent requests that happens very fast might get cancelled.